app/models/user.rb

```diff
@@ -66,7 +68,11 @@ def multiple_identities?
   end
 
   def active_profile
-    profiles.where(active: true).first
+    profiles.find(&:active?)
+  end
+
+  def verified?
+    active_profile ? true : false
   end
```

Comment:
I'm game to put business logic in its own class. There's lots of stuff on the `User` model currently that should go there. IMO:

```
  def need_two_factor_authentication?(request)
  def third_party_authenticated?(request)
  def two_factor_enabled?
  def send_two_factor_authentication_code(code)
  def confirmation_period_expired?
  def send_reset_confirmation
  def first_identity
  def last_identity
  def active_identities
  def multiple_identities?
  def active_profile
  def verified?
  def send_devise_notification(notification, *args)
```

are all candidates.

I just wouldn't call that business logic class a decorator. That name implies (to me anyway) something about presentation, not about acting upon or checking the state of a User.